test/serverTest.js

```diff
@@ -50,3 +53,153 @@ describe('server', () => {
       });
   });
 });
+
+describe('View Count', () => {
+  it('should update view count when progress equals one', (done) => {
+    const movieId = Math.floor(Math.random() * 300000);
+    const getViews = `select views from movies where id = ${movieId}`;
+    let viewsBefore = 0;
+    let viewsAfter = 0;
+    postgresDb.movieDb.query(getViews)
+      .then((resultBefore) => {
+        viewsBefore = resultBefore[0][0].views;
+        chai.request('http://localhost:3000')
+          .post('/tetraflix/sessionData')
+          .set('content-type', 'application/json')
+          .send({
+            userId: 534356757834,
+            events: [{
+              movie: {
+                id: movieId,
+              },
+              progress: 1,
+              timestamp: '2017-09-08 12:50PM',
+            }],
+          })
+          .end((err, res) => {
+            if (err) {
+              done(err);
+            }
+            postgresDb.movieDb.query(getViews)
+              .then((resultAfter) => {
+                viewsAfter = resultAfter[0][0].views;
+                viewsAfter.should.equal(viewsBefore + 1);
+              });
+          });
+        done();
+      });
+  });
+
+  it('should not update view count when progress is less than 1', (done) => {
+    const movieId = Math.floor(Math.random() * 300000);
+    const getViews = `select views from movies where id = ${movieId}`;
+    let viewsBefore = 0;
+    let viewsAfter = 0;
+    postgresDb.movieDb.query(getViews)
+      .then((resultBefore) => {
+        viewsBefore = resultBefore[0][0].views;
+        chai.request('http://localhost:3000')
+          .post('/tetraflix/sessionData')
+          .set('content-type', 'application/json')
+          .send({
+            userId: 12345,
```

Comment:
Are these userId's that you know to already exist in your database? Assuming they are, why is this one 5 digits, while the one on line 70 is 12 digits? Does this mean that all of the userIds in your database are of variable digit lengths?

Reply:
Yes, these are already existing users.  They all have variable digit lengths.